kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/SnapshotManager.java

```diff
@@ -475,6 +513,21 @@ protected Optional<LogSegment> getLogSegmentForVersion(
                 Arrays.toString(deltaVersionsAfterCheckpoint.toArray())
             ));
 
+        final long newVersion = deltaVersionsAfterCheckpoint.isEmpty() ?
+            newCheckpointOpt.get().version : deltaVersionsAfterCheckpoint.getLast();
```

Comment:
newCheckpointOpt can still be empty right? should we just use the `newCheckpointVersion`?

Reply:
Oh, is it because `deltaVersionsAfterCheckpoint` is empty only when the checkpoint is present? The checks in the beginning take care case where there is no checkpoint and delta files?